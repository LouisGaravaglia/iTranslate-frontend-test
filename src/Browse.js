import React from 'react';
import {useSpring, animated} from 'react-spring';
import './App.css';
import Categories from "./BrowseCategories";

function Browse() {

////////////////////////////////////////////////////  SPRING PROPS FOR BACKGROUND COLOR ANIMATION  ////////////////////////////////////////////////////

  const springProps = useSpring({
    backgroundColor: "#4e1eff",
    config: {duration: 300}
  });

////////////////////////////////////////////////////  JSX VARIABLES  ////////////////////////////////////////////////////

  //DISPLAY THE THREE CATEGORIES
  let ChooseCategoryDiv = (
    <animated.div style={springProps}>
      <Categories needAnimation={true}/>
    </animated.div>
  );

////////////////////////////////////////////////////  RETURN  ////////////////////////////////////////////////////

  return (
    <div>
      {ChooseCategoryDiv}
    </div>
  );
};

export default Browse;