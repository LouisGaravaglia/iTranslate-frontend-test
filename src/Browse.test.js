import React from 'react'
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import {Provider} from 'react-redux'
import {createStore} from "redux";
import Browse from "./Browse";

afterEach(cleanup);

const startingState = {};

/**
 * reducer function to mimic the reducer used in Redux.
 * @param {object} state - holds an object of data used within the components
 * @param {object} action - object where the values are strings used to fire certain actions
 */
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_ARTISTS":
      return {...state, artists: action.artists};
    default:
      return state;
  };
};

/**
 * Function that renders the component within a redux environment since most 
 * components rely on redux for data in order to mount.
 * @param {ReactComponent} component - private spotify client id
 * @param {function} store - creates the react store using the createStore method imported from redux
 */
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for Browse component', () => {

  it('renders without crashing', () => {
    renderWithRedux(<Browse />);
  });
});

//SNAPSHOT TEST
describe('Snapshot Test for Browse component', () => {

  it('matches snapshot', () => {
    const {asFragment} = renderWithRedux(<Browse />);
    expect(asFragment()).toMatchSnapshot();
  });
});