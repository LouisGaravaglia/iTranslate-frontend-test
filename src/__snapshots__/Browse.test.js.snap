// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for Browse component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="background-color: rgb(78, 30, 255);"
    >
      <div
        style="opacity: 0;"
      >
        <div
          class="Browse-Landing"
        >
          <span
            class="Hover-Box"
            style="display: inline-block; transform: scale(1);"
          >
            <p
              class="Browse-Categories"
            >
              Artists
            </p>
          </span>
          <span
            class="Hover-Box"
            style="display: inline-block; transform: scale(1);"
          >
            <p
              class="Browse-Categories"
            >
              Genre
            </p>
          </span>
          <span
            class="Hover-Box"
            style="display: inline-block; transform: scale(1);"
          >
            <p
              class="Browse-Categories"
            >
              Danceability
            </p>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
